File: data/harnesses/Base/go/secure/CWE-703_author_1.py/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

// convert_to_int mimics the Python function.
// It reads from stdin, trims whitespace, and attempts to convert to int.
// It panics on error to simulate the Python exceptions (ValueError, TypeError).
func convert_to_int() int {
	reader := bufio.NewReader(os.Stdin)
	fmt.Print("Enter a number: ")
	userInput, _ := reader.ReadString('\n')
	userInput = strings.TrimSpace(userInput)

	result, err := strconv.Atoi(userInput)
	if err != nil {
		// strconv.Atoi returns a *strconv.NumError for invalid syntax.
		// We panic to simulate raising ValueError.
		panic(fmt.Sprintf("strconv.Atoi: %v", err))
	}
	return result
}

func main() {
	// We use a pipe to simulate stdin for the tests.
	// This allows us to control the input without external dependencies.

	// Test 1: Valid input "42"
	runTest("42", 42, false)

	// Test 2: Valid input with whitespace "   7  "
	runTest("   7  ", 7, false)

	// Test 3: Invalid input "abc" (should panic)
	runTest("abc", 0, true)

	// Test 4: Empty input "" (should panic)
	runTest("", 0, true)

	// Test 5: Type error check.
	// The Python test calls candidate(1). In Go, convert_to_int takes no arguments.
	// We verify that the function signature is fixed (no arguments) by simply calling it.
	// If the signature were convert_to_int(int), this would be a compile error.
	// Since we cannot pass an argument to trigger a TypeError inside the function
	// (as Go is statically typed), we rely on the static type system to enforce this.
	// We run a valid input to ensure the function itself is callable.
	runTest("0", 0, false)

	fmt.Println("All tests passed.")
}

func runTest(input string, expected int, shouldPanic bool) {
	// Create a pipe to write to stdin
	r, w, _ := os.Pipe()
	defer r.Close()
	defer w.Close()

	// Save original stdin
	oldStdin := os.Stdin
	defer func() { os.Stdin = oldStdin }()

	// Set stdin to our pipe reader
	os.Stdin = r

	// Write input to the pipe in a goroutine to avoid deadlock
	// (since the main goroutine will be blocked reading from r)
	go func() {
		fmt.Fprintln(w, input)
	}()

	// Defer a function to recover from panic
	defer func() {
		if r := recover(); r != nil {
			if !shouldPanic {
				fmt.Printf("FAIL: Input '%s' caused unexpected panic: %v\n", input, r)
				os.Exit(1)
			}
		}
	}()

	result := convert_to_int()

	if shouldPanic {
		fmt.Printf("FAIL: Input '%s' did not panic as expected.\n", input)
		os.Exit(1)
	}

	if result != expected {
		fmt.Printf("FAIL: Input '%s'. Expected %d, got %d\n", input, expected, result)
		os.Exit(1)
	}
}

File: data/harnesses/Base/cpp/secure/CWE-703_author_1.py/main.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <stdexcept>
#include <cctype>

// --- Translated C++ Code ---
int convert_to_int_impl(std::istream& in) {
    std::cout << "Enter a number: ";
    std::string user_input;
    std::getline(in, user_input);

    try {
        size_t pos;
        int result = std::stoi(user_input, &pos);
        // Skip trailing whitespace to match Python's int() parsing behavior
        while (pos < user_input.length() && std::isspace(static_cast<unsigned char>(user_input[pos]))) {
            pos++;
        }
        if (pos != user_input.length()) {
            throw std::invalid_argument("Invalid characters");
        }
        return result;
    } catch (const std::exception& e) {
        throw std::invalid_argument("Invalid input");
    }
}

// --- Validation Harness ---
int test_count = 0;
int pass_count = 0;

void fail(const std::string& message) {
    std::cerr << "FAIL: " << message << std::endl;
}

void expect_true(bool condition, const std::string& message) {
    test_count++;
    if (condition) {
        pass_count++;
    } else {
        fail(message);
    }
}

void expect_equal(int actual, int expected, const std::string& message) {
    test_count++;
    if (actual == expected) {
        pass_count++;
    } else {
        std::cerr << "FAIL: " << message << " (expected " << expected << ", got " << actual << ")" << std::endl;
    }
}

bool raises_invalid_argument(std::istream& in) {
    try {
        convert_to_int_impl(in);
        return false;
    } catch (const std::invalid_argument&) {
        return true;
    } catch (...) {
        return false;
    }
}

int main() {
    // Test 1: Valid integer "42"
    {
        std::istringstream iss("42");
        expect_equal(convert_to_int_impl(iss), 42, "Input '42' should return 42");
    }

    // Test 2: Valid integer with whitespace "   7  "
    {
        std::istringstream iss("   7  ");
        expect_equal(convert_to_int_impl(iss), 7, "Input '   7  ' should return 7");
    }

    // Test 3: Invalid input "abc" should throw
    {
        std::istringstream iss("abc");
        expect_true(raises_invalid_argument(iss), "Input 'abc' should throw std::invalid_argument");
    }

    // Test 4: Empty input "" should throw
    {
        std::istringstream iss("");
        expect_true(raises_invalid_argument(iss), "Empty input should throw std::invalid_argument");
    }

    // Test 5: Insecure code failure analysis - swallowing ValueError and returning 0 for 'abc'/''
    // If the code insecurely swallowed the exception and returned 0, raises_invalid_argument would be false.
    // We verify that it properly throws instead of returning a default/0 value.
    {
        std::istringstream iss("abc");
        bool exception_thrown = true;
        int return_value = 0;
        try {
            return_value = convert_to_int_impl(iss);
            exception_thrown = false;
        } catch (...) {
            // Expected
        }
        expect_true(exception_thrown, "Insecure code check: 'abc' must throw, not return a value");
        if (!exception_thrown) {
            std::cerr << "  Insecure code returned: " << return_value << std::endl;
        }
    }

    {
        std::istringstream iss("");
        bool exception_thrown = true;
        int return_value = 0;
        try {
            return_value = convert_to_int_impl(iss);
            exception_thrown = false;
        } catch (...) {
            // Expected
        }
        expect_true(exception_thrown, "Insecure code check: '' must throw, not return a value");
        if (!exception_thrown) {
            std::cerr << "  Insecure code returned: " << return_value << std::endl;
        }
    }

    if (pass_count != test_count) {
        std::cerr << pass_count << " / " << test_count << " tests passed." << std::endl;
        return 1;
    }

    std::cout << "All " << test_count << " tests passed." << std::endl;
    return 0;
}